src/sentry/monitors/utils.py

```diff
@@ -383,3 +389,24 @@ def update_issue_alert_rule(
         )
 
     return issue_alert_rule.id
+
+
+def ensure_cron_detector(monitor: Monitor):
```

Comment:
I don't feel like much can go wrong here, but if we want to be more cautious we could put this creation behind a flag.